src/grid/onedgrid.py

```diff
@@ -384,22 +382,17 @@ def RectangleRuleSine(npoints):
     idx = np.arange(npoints) + 1
     points = (2 * idx - 1) / (2 * npoints)
 
-    weights = np.zeros(npoints)
+    weights = (
+        (2 / (npoints * np.pi ** 2))
+        * np.sin(npoints * np.pi * points)
+        * np.sin(npoints * np.pi / 2) ** 2
+    )
 
-    index_m = np.arange(npoints - 1) + 1
+    m = np.arange(npoints - 1) + 1
 
     for i in range(0, npoints):
-        elements = np.zeros(npoints - 1)
-        elements = np.sin(index_m * np.pi * points[i])
-        elements *= np.sin(index_m * np.pi / 2) ** 2
-        elements /= index_m
-
-        weights[i] = (4 / (npoints * np.pi)) * np.sum(elements)
-
-        weights[i] += (
-            (2 / (npoints * np.pi ** 2))
-            * np.sin(npoints * np.pi * points[i])
-            * np.sin(npoints * np.pi / 2) ** 2
+        weights[i] += (4 / (npoints * np.pi)) * np.sum(
+            np.sin(m * np.pi * points[i]) * np.sin(m * np.pi / 2) ** 2 / m
```

Comment:
Here is also the place where you could use vectorization skill to get rid of ``for`` loop.